article-recommendations-widget.php

```diff
@@ -92,31 +132,77 @@ public function widget($args, $instance) {
 		/**
 		 * Parse JSON response recommended articles and construct links to be rendered
 		 * Send data in the data-vars to be pulled in through AMP analytics
-		 *
+		 * {
+		 *	"results": [
+		 *		{
+		 *			"id": 9516554,
+		 *			"created_at": "2021-05-07T14:44:54.760391+00:00",
+		 *			"updated_at": "2021-05-07T14:44:54.760393+00:00",
+		 *			"source_entity_id": "default",
+		 *			"model": {
+		 *				"id": 625,
+		 *				"created_at": "2021-05-07T14:44:54.754762+00:00",
+		 *				"updated_at": "2021-05-07T14:44:54.852580+00:00",
+		 *				"type": "popularity",
+		 *				"status": "current"
+		 *			},
+		 *			"recommended_article": {
+		 *				"id": 4550,
+		 *				"created_at": "2021-04-15T00:20:21.728914+00:00",
+		 *				"updated_at": "2021-04-15T00:20:21.728927+00:00",
+		 *				"external_id": 286326,
+		 *				"title": "Lost Highway",
+		 *				"path": "/article/286326/lost-highway/",
+		 *				"published_at": "1996-09-27T04:00:00+00:00"
+		 *			},
+		 *			"score": "0.888446"
+		 *		}
+		 *	]
+		 * }
 		 * @param [type] $results
 		 * @return void
 		 */
 		function lnl_get_recommendations($results) {
+			$id = $results[0]['model']['id'];
+			$type = $results[0]['model']['type'];
+			$status = $results[0]['model']['status'];
+			$one = $results[0]['recommended_article']['external_id'];
+			$two=$results[1]['recommended_article']['external_id'];
+			$three=$results[2]['recommended_article']['external_id'];
+			$four=$results[3]['recommended_article']['external_id'];
+			$five=$results[4]['recommended_article']['external_id'];
+			$str = "data-vars-model-id=$id ";
+			$str .= "data-vars-model-status=$status ";
+			$str .= "data-vars-model-type=$type ";
+			$str .= "data-vars-one=$one ";
+			$str .= "data-vars-two=$two ";
+			$str .= "data-vars-three=$three ";
+			$str .= "data-vars-four=$four ";
+			$str .= "data-vars-five=$five ";
+
+			echo "<ul id='recommendations' " . $str . ">";
```

Comment:
since echoing has been iffy for us, does it make sense for us to build a string of html in `lnl_get_recommendations` and `lnl_get_recent_posts`, return that from each and then echo those where the functions are called, ie:
```
$recs = lnl_get_recommendations($results);		
echo $recs;		
$recent_posts = lnl_get_recent_posts();		
echo $recent_posts;
```
^ idk php so that syntax could very well be off

Reply:
I'm not sure, could try to fix in next iteration